engine/vcs/gitlab/client_status.go

```diff
@@ -81,17 +81,30 @@ func (c *gitlabClient) SetStatus(ctx context.Context, event sdk.Event) error {
 		Description: &data.desc,
 	}
 
-	if _, _, err := c.client.Commits.SetCommitStatus(data.repoFullName, data.hash, opt); err != nil {
-		return sdk.WrapError(err, "Cannot process event %v - repo:%s hash:%s", event, data.repoFullName, data.hash)
+	val, _, err := c.client.Commits.GetCommitStatuses(data.repoFullName, data.hash, nil)
+	if err != nil {
+		return sdk.WrapError(err, "Unable to get commit statuses %s", data.hash)
 	}
 
+	found := false
+	for _, s := range val {
+ 		if (s.TargetURL == *opt.TargetURL && s.Status == string(opt.State) && s.Ref == *opt.Ref && s.SHA == data.hash && s.Name == *opt.Name && s.Description == *opt.Description) {
```

Comment:
this is ugly, you should add comment or split on multiple lines, or use intermediate variables